### ISSUE TITLE
SCRAM-SHA-1 authentication failed after useDb

### ISSUE THREAD

**Issue by elgubenis** [Automattic/mongoose#3871]
i connect as following to my database:

```yml
uri: mongodb://hostname.tld:27017
options:
  server:
    poolSize: 50
    auto_reconnect: true
  user: lgubenis
  pass: password
  auth:
    db: admin
    authSource: admin
```

with a user which has readWriteAnyDatabase
```js
{
		"_id" : "admin.lgubenis",
		"user" : "lgubenis",
		"db" : "admin",
		"roles" : [
			{
				"role" : "readWriteAnyDatabase",
				"db" : "admin"
			}
		]
	}

```

when i first start my api, and mongoose connects, everything works as expected,
also when i use useDb on the connection for the first time, everything works fine, but the second time i use useDb i get the following error in mongodb: (switching db with useDb)

```sh
2016-02-10T16:17:35.782+0000 I ACCESS   [conn14] SCRAM-SHA-1 authentication failed for lgubenis on admin from client 169.57.121.196 ; BadValue Incorrect number of arguments for second SCRAM-SHA-1 client message, got 0 expected 3
2016-02-10T16:17:36.160+0000 I ACCESS   [conn14] Successfully authenticated as principal lgubenis on admin
2016-02-10T16:17:36.469+0000 I ACCESS   [conn14] SCRAM-SHA-1 authentication failed for lgubenis on admin from client 169.57.121.196 ; BadValue Unmatched SCRAM-SHA-1 nonce received from client in second step, expected v8ioKLxHjUUtc6uE7pHeNQV3kf+P4/jCwdl66Km3BabdOn/5AmaRkqvP5w/cNz8h but received Sr4Z5qugjsK2DOd8+dRr+s4/0lZHK9wjnAtjTbXdZBfo2dBrW2G6ps623QPv+fD/
2016-02-10T16:17:36.938+0000 I ACCESS   [conn14] Successfully authenticated as principal lgubenis on admin
2016-02-10T16:17:37.241+0000 I ACCESS   [conn14] SCRAM-SHA-1 authentication failed for lgubenis on admin from client 169.57.121.196 ; BadValue Unmatched SCRAM-SHA-1 nonce received from client in second step, expected 34kKZKFfBQJQqO6fOv82ibZSKwgoWG5zx+6wjEoBT89FS35j42QX6N34Wd3/XJ7v but received BnJCiAicWJ3DIeVBYa0/gcBjTUIrm1vQTyt9CkaQbX8k979QXm8i+jL3tbf+XMS/
2016-02-10T16:17:37.705+0000 I ACCESS   [conn14] Successfully authenticated as principal lgubenis on admin
2016-02-10T16:17:39.350+0000 I ACCESS   [conn14] SCRAM-SHA-1 authentication failed for lgubenis on admin from client 169.57.121.196 ; BadValue Incorrect number of arguments for second SCRAM-SHA-1 client message, got 0 expected 3
```

what am i doing wrong? what can i do better/right?
this worked with mongo 2.6, because we used another auth scheme, butt seems like with SCRAM-SHA-1 it does not work as expected (like the nonce expected is from another db or idk

**Comment by elgubenis** [comment]
sorry for the butt in but

**Comment by elgubenis** [comment]
if i do one query at a time everything works fine as it seems, but when i do 2 queries at once (2 useDb at the same time) the trouble seems to start.

my poolSize is 50, or 100, or 5

**Comment by elgubenis** [comment]
i just saw that i have the same problem with mongodb 2.6, when i deactivate auth (in mongod conf) i dont have problems anymore (but i also dont have security..) when i turn on auth i can only have 1 authd connection at a time it seems, any advice?

**Comment by vkarpov15** [comment]
That's an issue with mongoose, looks like we were attempting to re-authenticate when calling `useDb()`, which isn't the correct behavior because useDb() reuses the same connection under the hood, so it doesn't need to authenticate again. This fix will be in 4.4.4.
